src/v/cluster/rm_stm.cc

```diff
@@ -1812,15 +1809,62 @@ ss::future<raft::stm_snapshot> rm_stm::do_take_local_snapshot(
             stm_snapshot.producers.push_back(std::move(snapshot));
         }
     }
-    auto snapshot_offset = last_applied_offset();
+
     apply_units.return_all();
 
+    co_await ss::max_concurrent_for_each(
+      std::move(aborted_snapshots), 64, [this](abort_snapshot& snapshot) {
+          return save_abort_snapshot(std::move(snapshot));
+      });
+
+    _aborted_tx_state.abort_indexes = std::move(final_abort_indexes);
+    fragmented_vector<tx::tx_range> cleaned_aborted_ranges;
+    cleaned_aborted_ranges.reserve(_aborted_tx_state.aborted.size());
+    // preserve those aborted ranges which were not included in the snapshot
+    std::copy_if(
+      _aborted_tx_state.aborted.begin(),
+      _aborted_tx_state.aborted.end(),
+      std::back_inserter(cleaned_aborted_ranges),
+      [&ranges_offloaded_to_abort_snapshots](const tx_range& range) {
+          // only preserve ranges that were not included in the snapshot
+          return !ranges_offloaded_to_abort_snapshots.contains(range);
+      });
+    vlog(
+      _ctx_log.debug,
+      "Preserved {} aborted transaction ranges",
+      cleaned_aborted_ranges.size());
+    _aborted_tx_state.aborted = std::move(cleaned_aborted_ranges);
+    stm_snapshot.abort_indexes = _aborted_tx_state.abort_indexes.copy();
```

Comment:
nit: doesn't matter for correctness, but I think we can copy `final_abort_indexes` while we are still under the apply lock to underscore that the snapshot state is consistent